backup_mongo.sh

```diff
@@ -9,9 +9,47 @@ set -e
 # MONGO_AUTH_DB => name of mongo authentication database
 # AZURE_SA => Azure Storage account name
 # AZURE_BLOB_CONTAINER => name of the azure storage blob container
+# AZURE_SHARE_NAME => name of the azure file share
 # AZURE_DESTINATION_KEY => azure storage account destination key
 # DB => mongo db to backup
 
+# check the mongo uri
+if [ -z "$MONGO_URI" ]; then
+  echo "Error: you must set the MONGO_URI environment variable"
+  exit 1
+fi
+
+# check the mongo db
+if [ -z "$DB" ]; then
+  echo "Error: you must set the DB environment variable"
+  exit 2
+fi
+
+# check the azure args
+if [ -z "$AZURE_SA" ] || [ -z "$AZURE_DESTINATION_KEY" ]; then
+  echo "Error: you must set all Azure storage account variables AZURE_SA and AZURE_DESTINATION_KEY"
+  exit 3
+fi
+
+# get the azure destination type and name
+if [ ! -z "${AZURE_BLOB_CONTAINER}" ]; then
+  AZURE_TYPE=blob
+  AZURE_CONTAINER_NAME=${AZURE_BLOB_CONTAINER}
+elif [ ! -z "${AZURE_SHARE_NAME}" ]; then
+  AZURE_TYPE=file
+  AZURE_CONTAINER_NAME=${AZURE_SHARE_NAME}
+else
+  echo "Error: you must set either AZURE_BLOB_CONTAINER or AZURE_SHARE_NAME"
+  exit 4
+fi
+
+# check the mongo auth params
+if [ -z "$MONGO_USERNAME" ] && [ -z "$MONGO_PASSWORD" ] && [ -z "$MONGO_AUTH_DB" ]; then
+  NO_AUTH=${NO_AUTH:-true}
```

Comment:
I assume NO_AUTH was being passed from the .env before? if so it'll use that value or if not present set to true

Reply:
yes, it is passed from env (in a kubernetes cluster)